src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtlDocumentTransformer.cs

```diff
@@ -247,25 +247,28 @@ public override void Transform(RavenEtlItem item, EtlStatsScope stats, EtlProces
                     case EtlItemType.Document:
                         if (_script.HasTransformation)
                         {
-                            DocumentScript.Run(Context, Context, "execute", new object[] { Current.Document }).Dispose();
-                            ApplyDeleteCommands(item, OperationType.Put, out var isLoadedToDefaultCollectionDeleted);
-
-                            if (_currentRun.IsDocumentLoadedToSameCollection(item.DocumentId) == false)
-                                break;
-                            
-                            if (_script.TryGetLoadCounterBehaviorFunctionFor(item.Collection, out var counterFunction))
-                            {
-                                var counterGroups = GetCounterGroupsFor(item);
-                                if (counterGroups != null)
-                                    AddCounters(item.DocumentId, counterGroups, counterFunction);    
-                            }
-                            if (_script.TryGetLoadTimeSeriesBehaviorFunctionFor(item.Collection, out var timeSeriesLoadBehaviorFunc))
+                            using (DocumentScript.Run(Context, Context, "execute", new object[] {Current.Document}))
```

Comment:
Note: the fix is here that instead of calling `DocumentScript.Run().Dispose()` immediately we wrap it by `using()`